centrallix/osdrivers/objdrv_sybase.c

```diff
@@ -795,29 +797,16 @@ sybd_internal_GetCxValue(void* ptr, int ut, pObjData val, int datatype)
 		    if (lsl == 0xFFFFFFFF) msl++;
 		    lsl++;
 		    }
-		/** Long division, 16 bits = 1 "digit" **/
-		divtmp = msl/10000;
-		n = divtmp;
-		msl -= divtmp*10000;
-		msl = (msl<<16) + (lsl>>16);
-		divtmp = msl/10000;
-		n = (n<<16) + divtmp;
-		msl -= divtmp*10000;
-		msl = (msl<<16) + (lsl & 0xFFFF);
-		divtmp = msl/10000;
-		n = (n<<16) + divtmp;
-		msl -= divtmp*10000;
-		val->Money->WholePart = n;
-		val->Money->FractionPart = msl;
+		/** Both msl and lsl are little endian, so I want lsl to come before msl.
+		** I do this by shifting lsl left in a long long (ll_lsl) by 32 bits, and then OR it with msl
+		**/
+		ll_msl = msl;
+		ll_lsl = lsl;
+		ll_lsl = ll_lsl << 32;
+		ll_lsl = ll_lsl | ll_msl;
+		val->Money->Value = ll_lsll;
```

Comment:
Undefined symbol ll_lsll here (I think you meant ll_lsl), also msl is the most significant 32-bit value, and should be left shifted instead of lsl.  I think you can just do val->Money->Value = (((long long)msl)<<32) | (long long)lsl;